scripts/data_prep/convert_delta_to_json.py

```diff
@@ -0,0 +1,451 @@
+            decompressed_data = lz4.frame.decompress(resp.content)
+            # Convert the decompressed data into a PyArrow table
+            reader = pa.ipc.open_stream(decompressed_data)
+        else:
+            reader = pa.ipc.open_stream(resp.content)
+        table = reader.read_all()
+
+        # Convert the PyArrow table into a pandas DataFrame
+        df = table.to_pandas()
+        df.to_json(os.path.join(json_output_path,
+                                'part_' + str(ipart) + '.jsonl'),
+                   orient='records',
+                   lines=True,
+                   force_ascii=False)
+
+
+def download_starargs(args: Tuple) -> None:
+    return download(*args)
+
+
+def fetch_data(method: str, cursor: Optional[Cursor],
+               sparkSession: Optional[SparkSession], s: int, e: int,
+               order_by: str, tablename: str, columns_str: str,
+               json_output_path: str) -> None:
+    query = f"""
+    WITH NumberedRows AS (
+        SELECT
+            *,
+            ROW_NUMBER() OVER (ORDER BY {order_by}) AS rn
+        FROM
+            {tablename}
+    )
+    SELECT {columns_str}
+    FROM NumberedRows
+    WHERE rn BETWEEN {s+1} AND {e}"""
+
+    if method == 'dbconnect':
+        spark_df = run_query(query, method, cursor, sparkSession, collect=False)
+        if not spark_df:
+            raise RuntimeError(
+                f'Expect spark dataframe with {query} but got None')
+        pdf = spark_df.toPandas()  # pyright: ignore
+    else:  #  method == 'dbsql':
+        ans = run_query(query, method, cursor, sparkSession, collect=True)
+        if not ans:
+            raise RuntimeError(f'Got empty results with {query}')
+        records = [r.asDict() for r in ans]  # pyright: ignore
+        pdf = pd.DataFrame.from_dict(records)
+
+    pdf.to_json(os.path.join(json_output_path, f'part_{s+1}_{e}.jsonl'))
+
+
+def fetch(
+    method: str,
+    tablename: str,
+    json_output_path: str,
+    batch_size: int = 1 << 30,
+    processes: int = 1,
+    sparkSession: Optional[SparkSession] = None,
+    dbsql: Optional[Connection] = None,
+) -> None:
+    """Fetch UC delta table with databricks-connnect as JSONL.
+
+    Args:
+        method (str): dbconnect or dbsql
+        tablename (str): catalog.scheme.tablename on UC
+        json_output_path (str): path to write the result json file to
+        batch_size (int): number of rows that dbsql fetches each time to avoid OOM
+        processes (int): max number of processes to use to parallelize the fetch
+        sparkSession (pyspark.sql.sparksession): spark session
+        dbsql (databricks.sql.connect): dbsql session
+    """
+    cursor = dbsql.cursor() if dbsql is not None else None
+
+    try:
+        ans = run_query(f'SELECT COUNT(*) FROM {tablename}', method, cursor,
+                        sparkSession)
+        nrows = [row.asDict() for row in ans][0].popitem()[1]  # pyright: ignore
+        log.info(f'total_rows = {nrows}')
```

Comment:
not sure if we need to log this as `info` or `debug`. Would it help user when they see this information?

Reply:
``debug`` is more appropriate.